xbmc/Util.cpp

```diff
@@ -2467,3 +2468,19 @@ void CUtil::CopyUserDataIfNeeded(const std::string& strPath,
     CFile::Copy(srcPath, destPath);
   }
 }
+
+bool CUtil::FileDelWithConfirm(CFileItemPtr pItem)
+{
+  CGUIComponent* gui = CServiceBroker::GetGUI();
+
+  if (gui)
+  {
+    const bool confirm = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+        CSettings::SETTING_FILELISTS_CONFIRMFILEDELETE);
+
+    if (!confirm || gui->ConfirmDelete(pItem->GetPath()))
+      return CFileUtils::DeleteItem(pItem);
+  }
+
+  return false;
```

Comment:
No need to check for `gui` presence first. Settings manager works also without GUI. Should read, like:

```c++
bool deleteItem = false;

if (CServiceBroker::GetSettingsComponent()->GetS...
{
  CGUIComponent* gui = CService...
  if (gui)
    deleteItem = gui->Confirm...
}
else
{
  deleteItem = true;
}

if (deleteItem)
 return CFileUtil::DeleteItem...
 
return false;
```